microcosm_daemon/reloader.py

```diff
@@ -0,0 +1,82 @@
+"""
+Daemon reloading on code change (for development/debug).
+
+Cribbed in part from: https://github.com/pallets/werkzeug/blob/master/werkzeug/_reloader.py
+
+"""
+from logging import getLogger
+import os
+import sys
+
+
+def _iter_module_files():
+    """This iterates over all relevant Python files.  It goes through all
+    loaded files from modules, all files in folders of already loaded modules
+    as well as all files reachable through a package.
+    """
+    # The list call is necessary on Python 3 in case the module
+    # dictionary modifies during iteration.
+    for module in list(sys.modules.values()):
+        if module is None:
+            continue
+        filename = getattr(module, '__file__', None)
+        if filename:
+            old = None
+            while not os.path.isfile(filename):
+                old = filename
+                filename = os.path.dirname(filename)
+                if filename == old:
+                    break
+            else:
+                if filename[-4:] in ('.pyc', '.pyo'):
+                    filename = filename[:-1]
+                yield filename
+
+
+class Reloader(object):
+    """
+    Checks modification times for Python modules and reloads execution.
+
+    """
+    def __init__(self):
+        self.mtimes = {}
+
+    @property
+    def changed(self):
```

Comment:
This is very similar to the equivalent code in `werkzeug` except that I wrote it as a `@property` instead of a loop since the daemon infrastructure already has its own loop.